src/SignalR/clients/csharp/Client/test/FunctionalTests/Startup.cs

```diff
@@ -49,6 +51,11 @@ public void ConfigureServices(IServiceCollection services)
                                 IssuerSigningKey = SecurityKey
                             };
                     });
+
+            // Since tests run in parallel, it's possible multiple servers will startup and read files being written by another test
+            // Use a unique directory per server to avoid this collision
+            services.AddDataProtection()
+                .PersistKeysToFileSystem(Directory.CreateDirectory(Path.GetRandomFileName()));
```

Comment:
Is this directory created in the CI agent's working directory so it's cleaned up automatically?

Reply:
It was locally. Haven't looked at the CI logs yet.

Reply:
DataProtection file placed at `/datadisks/disk1/work/B72C09B7/w/A7CD0956/e/gx2nq0by.eg5/key-423cc81a-aa77-44aa-958b-f873f764b388.xml` and the working directory is `/datadisks/disk1/work/B72C09B7/w/A7CD0956/e`